src/platform_impl/linux/taskbar/xapps.rs

```diff
@@ -0,0 +1,162 @@
+use std::ffi::CStr;
+
+use lazy_static::lazy_static;
+use raw_window_handle::{
+  RawDisplayHandle, RawWindowHandle, XcbDisplayHandle, XcbWindowHandle, XlibDisplayHandle,
+  XlibWindowHandle,
+};
+use x11_dl::xlib::{self, PropModeReplace, Xlib};
+use x11rb::{
+  protocol::xproto::{AtomEnum, ConnectionExt, PropMode},
+  wrapper::ConnectionExt as WrapperConnectionExt,
+  xcb_ffi::XCBConnection,
+};
+
+lazy_static! {
+  static ref XLIB: Xlib = Xlib::open().unwrap();
+}
+
+pub struct Manager {
+  handle: (RawWindowHandle, RawDisplayHandle),
+  progress: u32,
+  progress_visible: bool,
+  pulse: bool,
+}
+
+impl Manager {
+  pub fn new(handle: RawWindowHandle, d_handle: RawDisplayHandle) -> Option<Self> {
+    if !matches!(handle, RawWindowHandle::Xlib(_) | RawWindowHandle::Xcb(_)) {
```

Comment:
you should check for xlib handle only since that is the only one we create, we never create xcb handle